### DISCUSSION TITLE
Ranges

### DISCUSSION THREAD

**Discussion by cpurdy** [xtclang/xvm#40] [Ideas]
Swift uses `..` for ranges. Not surprising.

I just learned that they use `..<` _exclusive_ ranges. Ecstasy uses the math notation, e.g. `[inclusive..exclusive)`, which causes all sorts of IDE and editor confusion.

Curious if anyone has an opinion on this topic?

**Comment by ggleyzer** [comment]
I like it; while _(a..b]_ still looks to me a bit better than _a..<b_, the fact that you can do _a>..b_ and _a>..<b_ and keep the parenthesis balanced is a big plus. And then some expressions do look like emoticons, for example:

    (-w)>..<(w)
    (o)>..<(O)

**Comment by cpurdy** [comment]
Plan of record:

* Get rid of the `[x..y]` and `[x..y)` constructs e.g.

```
    for (Int i : 5..10)     // this will still exist 
    for (Int i : [5..10])   // this used to be the same behavior as line above, but this goes away
    for (Int i : [5..10))   // this is exclusive on the upper bound (10), but this goes away
```

* The old way could **not** do exclusive lower + exclusive upper bound:

```
    x === (x)         // basic expression guarantee
    a..b === (a..b)   // this follows the above rule, but the parenthesis implied "exclusive mode"
```

* Decision: adopt the "Swift like" operator style, aka "gremlin operators" (Mark Falco's  name for them)

```
    a  ..  b
    a >..  b
    a  ..< b
    a >..< b
```

**Comment by cpurdy** [comment]
OK, this one is all done and in. The old syntax has also been removed.

(The bootstrapped lexer / parser built in Ecstasy still needs to be updated similarly.)
